locust/runners.py

```diff
@@ -578,27 +586,33 @@ def start(self, user_count, spawn_rate):
                 data["num_users"] += 1
                 remaining -= 1
 
+            logger.debug("Sending spawn message to client %s" % (client.id))
             self.server.send_to_client(Message("spawn", data, client.id))
 
-        self.state = STATE_SPAWNING
+        self.update_state(STATE_SPAWNING)
 
     def stop(self):
         if self.state not in [STATE_INIT, STATE_STOPPED, STATE_STOPPING]:
-            self.state = STATE_STOPPING
+            logger.debug("Stopping...")
```

Comment:
oh, but I think this line is no longer needed. you can remove it as part of the actual fix later on :)

Reply:
Why is it not needed?

Reply:
Because we’re also going to be logging ”Updating state to STATE_CLEANUP, old state was X” right after that.